src/storage-types/src/sinks.rs

```diff
@@ -422,6 +448,81 @@ pub struct KafkaSinkConnection<C: ConnectionAccess = InlinedConnection> {
     pub connection_options: BTreeMap<String, StringOrSecret>,
 }
 
+impl<C: ConnectionAccess> KafkaSinkConnection<C> {
+    /// Determines if `self` is compatible with another `StorageSinkConnection`,
+    /// in such a way that it is possible to turn `self` into `other` through a
+    /// valid series of transformations (e.g. no transformation or `ALTER
+    /// CONNECTION`).
+    pub fn alter_compatible(
+        &self,
+        id: GlobalId,
+        other: &KafkaSinkConnection<C>,
+    ) -> Result<(), StorageError> {
+        if self == other {
+            return Ok(());
+        }
+        let KafkaSinkConnection {
+            connection_id,
+            // The details of the Kafka connection itself may change
+            connection: _,
+            format,
+            relation_key_indices,
+            key_desc_and_indices,
+            value_desc,
+            topic,
+            consistency_config,
+            partition_count,
+            replication_factor,
+            fuel,
+            retention,
+            connection_options,
+        } = self;
+
+        let compatibility_checks = [
+            (connection_id == &other.connection_id, "connection_id"),
+            (format.alter_compatible(id, &other.format).is_ok(), "format"),
+            (
+                relation_key_indices == &other.relation_key_indices,
+                "relation_key_indices",
+            ),
+            (
+                key_desc_and_indices == &other.key_desc_and_indices,
+                "key_desc_and_indices",
+            ),
+            (value_desc == &other.value_desc, "value_desc"),
+            (topic == &other.topic, "topic"),
```

Comment:
do we check the progress topic anywhere? (i could be missing something obvious)

Reply:
Yes, it's the consistency config